test/inductor/test_torchinductor.py

```diff
@@ -5203,6 +5203,31 @@ def test_layer_norm(self):
         if self.device != "cpu":
             assertGeneratedKernelCountEqual(self, 1)
 
+    def test_matmul_layer_norm(self):
```

Comment:
It might be nice in the future to have a way of writing tests for this just in the forward. donated buffer for forward inference is something people occasionally do. 

For now if we factored out `get_donated_idxs` we could patch it and that would give a way of testing.

Could we add test of the form :
```
def foo(x: donated_buffer):
     z = x + 1
     return (z @ x)
```

to make sure we don't inplace the `x + 1` kernel  here ?